changefeedccl: assert increasing resolved message timestamps in validators

jayshrivastava opened the pull request:
streamingccl: fix incorrect usages of cdctest.Validator

The `NoteResolved` API on the `cdctest.Validator` requires that the passed
resolved timestamp is the resolved timetsamp / frontier for the whole set
of spans being tracked by the validator. Previously, tests would submit resolved
timestamps for subsets of spans at a time, which is incorrect. This would
not cause tests to fail because the order validator does not
assert that resolved timetsamps monotonically increase (tracked by https://github.com/cockroachdb/cockroach/issues/108872).
This change updates these tests to submit resolved timestamps for the
overall set of spans being tracked. After this change, fixing the
ordering validator in (https://github.com/cockroachdb/cockroach/issues/108872) will not cause these tests to fail.

Release note: None
Epic: None

--- 

changefeedccl: assert increasing resolved message timestamps in validators #108861

Previously, no validator in the `cdctest` package would assert that resolved messages increase monotonically. This is an important ordering guarantee that changefeeds provide.

This change updates the `orderValidator` to assert this.

Release note: None
Epic: None
Fixes: https://github.com/cockroachdb/cockroach/issues/108872

jayshrivastava committed:
changefeedccl: assert increasing resolved message timestamps in validators

Previously, no validator in the `cdctest` package would assert that resolved
messages increase monotonically. This is an important ordering guarantee that
changefeeds provide.

This change updates the `orderValidator` to assert this.

Release note: None
Epic: None
Fixes: #108872

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/108861)
<!-- Reviewable:end -->

jayshrivastava committed:
streamingccl: fix incorrect usages of cdctest.Validator

The `NoteResolved` API on the `cdctest.Validator` requires that the passed
resolved timestamp is the resolved timetsamp / frontier for the whole set
of spans being tracked by the validator. Previously, tests would submit resolved
timestamps for subsets of spans at a time, which is incorrect. This would
not cause tests to fail because the order validator does not
assert that resolved timetsamps monotonically increase (tracked by https://github.com/cockroachdb/cockroach/issues/108872).
This change updates these tests to submit resolved timestamps for the
overall set of spans being tracked. After this change, fixing the
ordering validator in (https://github.com/cockroachdb/cockroach/issues/108872) will not cause these tests to fail.

Release note: None
Epic: None
